S/SZ/build_tarballs.jl

```diff
@@ -0,0 +1,104 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "SZ"
+version = v"2.1.12"
+version_string = "2.1.12.5"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/szcompressor/SZ/releases/download/v$(version_string)/SZ-$(version_string).tar.gz",
+                  "32a820daf6019156a777300389d2392e4498a5c9daffce7be754cd0a5ba8729c"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir
+cd SZ-*
+
+hdf5_options=
+if test -f "${includedir}/hdf5.h"; then
+    # HDF5 is available, use it
+    hdf5_options='-DBUILD_HDF5_FILTER=ON -DBUILD_NETCDF_READER=ON'
+else
+    # Create an empty library
+    echo 'int SZ_no_hdf5;' >hdf5sz.cxx
+    c++ -fPIC -c hdf5sz.cxx
+    c++ -shared -o libhdf5sz.${dlext} hdf5sz.o
+    install -Dvm 755 "libhdf5sz.${dlext}" "${libdir}/libhdf5sz.${dlext}"
+    # Create an empty library
+    echo 'int SZ_no_netcdf;' >netcdfsz.cxx
+    c++ -fPIC -c netcdfsz.cxx
+    c++ -shared -o libnetcdfsz.${dlext} netcdfsz.o
+    install -Dvm 755 "libnetcdfsz.${dlext}" "${libdir}/libnetcdfsz.${dlext}"
+fi
+
+openmp_options=
+if [[ "${target}" != *-apple-* ]]; then
+    openmp_options='-DBUILD_OPENMP=ON'
+fi
```

Comment:
Would https://github.com/JuliaPackaging/Yggdrasil/blob/7176f726e4a0628b72f15bfa2afeb9189a82d61e/B/blis/build_tarballs.jl#L135-L138 help?  macOS toolchain (also on real macOS) doesn't come with OpenMP at all, you need to bring it in from somewhere else.

Reply:
Thanks, I'll try this.